### PR TITLE
Table of comments

### PR DESCRIPTION
**Pull request by CristianRWolox**
## Summary

- Add table of comments to book details screen.

#### Galaxy S (or similar)
image

#### Pixel 2/3 (or similar)
image

#### Pixel 3XL (or similar)
image

#### iPhone SE (or similar)
simulator_screenshot_CBFAF5FC-C4D6-49F9-BD3E-4B01E99078BF

#### iPhone 6/7/8 (or similar)
simulator_screenshot_4E92B9CC-6EC6-496C-B5BE-956865820D6E

#### iPhone 11/12 Base/Pro/Pro Max (or similar)
image

## Trello Card

https://trello.com/c/eB3O2GGL/16-n-73-tabla-de-comentarios

### CODE REVIEW

**Review by mcolombo87 — COMMENTED** [review]
The "into <Branch>" must be **bookDetail**. I can see changes in this PR that aren't of this card.

File: wbooks/src/app/screens/BookDetail/components/Comments/index.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import user1 from '@assets/General/img_user1.png';
+import user2 from '@assets/General/img_user2.png';
+
+import styles from './styles';
+
+const Comments = () => {
+  const [viewAll, setViewAll] = useState(false);
+  return (
+    <View style={styles.commentCard}>
+      {COMMENTS_MOCK.filter(comment => (viewAll ? true : comment.id < 2)).map(comment => (
```
Comment (mcolombo87) [inline wbooks/src/app/screens/BookDetail/components/Comments/index.tsx]: Think of this as information provided by a backend service. This does not scale. Use a flat list and set a "Comment" component as the item to render. If you received 10 comments from the backend, this view doesn't even have a scroll.

File: wbooks/src/app/screens/BookDetail/components/Comments/index.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import React, { useState } from 'react';
+import { View, Text, TouchableOpacity, Image } from 'react-native';
+import { COMMENTS_MOCK } from '@constants/mockComments';
+import user1 from '@assets/General/img_user1.png';
```
Comment (mcolombo87) [inline wbooks/src/app/screens/BookDetail/components/Comments/index.tsx]: This must be inside of the "comment" mock. Is information provided by a service (We are mocking that data really, but think it like that)

**Review by mcolombo87 — CHANGES_REQUESTED** [review]

File: wbooks/src/app/screens/BookDetail/components/Comments/index.tsx (hunk 1 above)
Comment (mcolombo87) [reply wbooks/src/app/screens/BookDetail/components/Comments/index.tsx]: Ok, I see the scroll in Book Detail but I don't think if this is the best approach. If you want the scroll in the father don´t use the flatlist but this map is not the best. You can use a flat with a header, I will try this of another way... what do you think?

Comment (CristianRWolox) [reply wbooks/src/app/screens/BookDetail/components/Comments/index.tsx]: My idea is to use ScrollView - map, I thought of using flatlist but I think is an error put a FlatList inside a parent Scroll.

It's not clear for me the 'flat with a header'

What can I do about the flatlist, put it on the child and the parent with scroll?

File: wbooks/src/app/screens/BookDetail/components/Comments/index.tsx (hunk 2 above)
Comment (CristianRWolox) [reply wbooks/src/app/screens/BookDetail/components/Comments/index.tsx]: Sorry, it's not clear for me, What must be inside?

File: wbooks/src/app/screens/BookDetail/components/Comments/index.tsx (hunk 1 above)
Comment (mcolombo87) [reply wbooks/src/app/screens/BookDetail/components/Comments/index.tsx]: Use the scrollview or the flatlist, no both. But really really i don't like this approach. The flatlist have optimization and was builded it for this situations, i think is the better option. You can scroll only the comments if you consider that as an issue or use ListHeaderComponent and get the same behavior.

File: wbooks/src/app/screens/BookDetail/components/Comments/index.tsx (hunk 2 above)
Comment (mcolombo87) [reply wbooks/src/app/screens/BookDetail/components/Comments/index.tsx]: the mock data perhaps need an item like "userImage" and get that data from there.

Comment (guidoprinc) [reply wbooks/src/app/screens/BookDetail/components/Comments/index.tsx]: what he is saying that you should move this inside `mockComments` so when you get the real pictures, we will use only those

File: wbooks/src/app/screens/BookDetail/components/Comments/index.tsx (hunk 1 above)
Comment (guidoprinc) [reply wbooks/src/app/screens/BookDetail/components/Comments/index.tsx]: https://javascript.plainenglish.io/how-to-fix-virtualizedlists-should-never-be-nested-inside-plain-scrollviews-warning-3a2a887b4ea0

Check this post so you can add the information as the header and the comments as the real data so we avoid using a ScrollView and a FlatList

File: wbooks/src/app/screens/BookDetail/components/Comments/index.tsx (hunk 2 above)
Comment (CristianRWolox) [reply wbooks/src/app/screens/BookDetail/components/Comments/index.tsx]: Done!

File: wbooks/src/app/screens/BookDetail/components/Comments/index.tsx (hunk 1 above)
Comment (CristianRWolox) [reply wbooks/src/app/screens/BookDetail/components/Comments/index.tsx]: Done!

**Review by guidoprinc — APPROVED** [review]

**Review by mcolombo87 — APPROVED** [review]

**Pull request merged**